library/std/src/io/mod.rs

```diff
@@ -2989,7 +2989,7 @@ impl<T: Read> Read for Take<T> {
             return Ok(());
         }
 
-        if self.limit <= buf.capacity() as u64 {
+        if self.limit < buf.capacity() as u64 {
             // if we just use an as cast to convert, limit may wrap around on a 32 bit target
             let limit = cmp::min(self.limit, usize::MAX as u64) as usize;
```

Comment:
If you want, I think you could remove these two lines as well – if `self.limit` is smaller than `buf.capacity()` (which is a `usize`), it must be smaller than `usize::MAX` anyway.